sources.go

```diff
@@ -113,3 +113,19 @@ func Slice(s []string) *Pipe {
 func Stdin() *Pipe {
 	return NewPipe().WithReader(os.Stdin)
 }
+
+// Prompt presents a message to the user and returns a pipe containing user input or defaultValue provided.
+func Prompt(message, defaultValue string) *Pipe {
+
+	Echo(message).Stdout()
+	p := Stdin().First(1)
```

Comment:
Should we show a prompt character, perhaps, to indicate to the user that the program is waiting for input?

Reply:
We should indicate the default value to the user too, shouldn't we?

Reply:
The message now displays the Prompt message and the default value between [ ].

What do you mean by prompt character ? Something like a blinking text input ?

Reply:
That looks good! We should maybe leave a space after the colon before the user's input?

Reply:
Done